src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -169,53 +168,87 @@ void BuildingComponentDialogCentralWidget::setTid() {
   requestComponents(m_filterType, m_tid, m_pageIdx, m_searchString);
 }
 
+std::vector<openstudio::BCLSearchResult> BuildingComponentDialogCentralWidget::fetchAndSortResponses(const std::string& filterType, int tid, const QString& searchString) {
+  m_allResponses.clear();
+  RemoteBCL remoteBCL;
+  std::vector<BCLSearchResult> responses;
+  int totalPages = 1;
+  int currentPage = 0;
+
+  // Collect all responses from all pages
+  do {
+    std::vector<BCLSearchResult> pageResponses;
+    if (filterType == "components") {
+      pageResponses = remoteBCL.searchComponentLibrary(searchString.toStdString(), tid, currentPage);
+    } else if (filterType == "measures") {
+      pageResponses = remoteBCL.searchMeasureLibrary(searchString.toStdString(), tid, currentPage);
+    }
+    responses.insert(responses.end(), pageResponses.begin(), pageResponses.end());
+    totalPages = remoteBCL.numResultPages();
+  } while (++currentPage < totalPages);
+
+  if (!responses.empty()) {
+    std::sort(responses.begin(), responses.end(), [](const BCLSearchResult& a, const BCLSearchResult& b) {
+      return a.name() < b.name();
+    });
+  }
+
+  return responses;
+}
+
 // Note: don't call this directly if the "wait" screen is desired
 void BuildingComponentDialogCentralWidget::setTid(const std::string& filterType, int tid, int pageIdx, const QString& title,
                                                   const QString& searchString) {
 
-  if (m_tid != tid || m_searchString != searchString) {
-    m_collapsibleComponentList->firstPage();
-  }
+  std::string newKey = std::to_string(tid) + filterType + searchString.toStdString();
+  std::string currentKey = std::to_string(m_tid) + m_filterType + m_searchString.toStdString();
 
-  m_filterType = filterType;
+  m_collapsibleComponentList->firstPage();
```

Comment:
Do you only want to force navigating to the first page if the key changes?

Reply:
Yes it seems simplier to do that since that was the original functionality

Reply:
Updated